src/humanize/number.py

```diff
@@ -26,6 +26,18 @@
 NumberOrString: TypeAlias = "float | str"
 
 
+def format_non_finite(value: float) -> str:
+    """Utility function to handle infinite and nan cases."""
+    if math.isnan(value):
+        return "NaN"
+    if math.isinf(value) and value < 0:
+        return "-Inf"
+    elif math.isinf(value) and value > 0:
```

Comment:
You may be interested in https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-return.html :)